### ISSUE TITLE
[nextcloud] only ask password on relogin

### ISSUE THREAD

**Issue by ArnY** [jsxc/jsxc#489]
Since the user is already logged in nextcloud, I'm sure we can retrieve their username or email address(depending on the way the admin configured ojsxc) so only asking for the password could ease the (re)login process and make the integration much better.

**Comment by sualko** [comment]
That's a good idea. We are also currently working on a solution which doesn't even require a password prompt, see nextcloud/jsxc.nextcloud#4.

**Comment by ArnY** [comment]
I know about the token based solution, i've been following the discussion, but wouldn't that requires to also make change in the way the xmpp server does authentication? Wouldn't it be simpler to get the password from the login screen as if the user was about to be logged on and save it in a two way encrypted form in the session?

**Comment by fsedarkalex** [comment]
I would highly appreciate storing the password two-way-encrypted somewhere so thet a reload of the page does not require a relogin...

Currently every app change in NC does a full page reload.
Also in OC there are some Apps which cause a Page-Relaod and each relaod causes the user to be logge doff. This is simply unusable...

I am using openfire here btw...

**Comment by sualko** [comment]
> wouldn't that requires to also make change in the way the xmpp server does authentication?

Yes, but I wrote a external auth script for prosody and ejabberd.

> save it in a two way encrypted form in the session?

Yes it would be simpler, but also a huge security risk, because every javascript snippet (which is executed in nc) could read the password.

@fsedarkalex a reload should not result in a disconnect. Please check your javascript console and xmpp server log for errors and open a new issue.

**Comment by ArnY** [comment]
well, as soon as nextcloud/server/pull/2044 is merged, we could retrieve the user credentials from there

**Comment by fsedarkalex** [comment]
nextcloud/server#2044 is merged now. I noticed current version of JSXC on NC12 is a bit more stable but also loses connection after a while.
